### PR TITLE
Updated: build config

### PR DESCRIPTION
**Pull request by noob77777**
# Updates build files and read me

## Remaining tasks

- [ ] Fix Codecov after link change

## Further work

### CODE REVIEW

**Comment by TheSYNcoder** [comment]
Suggestion : 

Let's have one commit message per pull request. Can be done by ``` git commit --amend```
We can add this to the docs.

**Comment by noob77777** [comment]
> 
> 
> Suggestion :
> 
> Let's have one commit message per pull request. Can be done by ` git commit --amend`
> We can add this to the docs.

We will be defaulting to squash and merge.

**Comment by bisakhmondal** [comment]
> > Suggestion :
> > Let's have one commit message per pull request. Can be done by ` git commit --amend`
> > We can add this to the docs.
> 
> We will be defaulting to squash and merge.

Agree +1
with amend you can't cherry-pick

**Review by bisakhmondal — APPROVED** [review]
\lgtm

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      macro: no
+
+comment:
+  layout: "reach, diff, flags, files, footer"
+  behavior: default
+  require_changes: no
+
+ignore:
+  # ignore the third_party directory
+  - "third_party"
```
Comment (bisakhmondal) [inline codecov.yml:32]: Minor change
Add the following directories
```
build_support
script
docs
.github
```

**Review by bisakhmondal — CHANGES_REQUESTED** [review]

File: codecov.yml (hunk 1 above)
Comment (bisakhmondal) [reply codecov.yml:32]: cool then

**Pull request merged**